### PR TITLE
#47 - Refactored Renderer to enable nested loops

### PR DESCRIPTION
**Pull request by chikeichan**

### CODE REVIEW

**Comment by chikeichan** [comment]
resolves #47

**Comment by chikeichan** [comment]
resolves #47

**Comment by chikeichan** [comment]
resolves #47

**Comment by chikeichan** [comment]
closes #47

File: spec/renderer/renderer.spec.js [hunk 1]
```diff
@@ -17,8 +17,8 @@ describe('The Renderer Class', function() {
 
     it('should be able to add and remove class', function() {
```
Comment (sdtsui) [inline spec/renderer/renderer.spec.js]: 18 - suggest add/remove be tested separately
  -Possible edge cases:
    -what happens when you add a class that is already there?
    -what happens when you remove a class that is not there?

File: spec/renderer/renderer.spec.js [hunk 2]
```diff
@@ -40,34 +40,52 @@ describe('The Renderer Class', function() {
...
             .attr('contenteditable', 'true')
-         .appendLast();
+         .close();
         frag = t.render();
         el.appendChild(frag);
 
         expect(el.innerHTML).toBe('<input disabled="true"><div contenteditable="true"></div>');
     });
 
+    it('should be able to render multiple times', function() {
```
Comment (sdtsui) [inline spec/renderer/renderer.spec.js]: 55 - could use a stub for this  - count if called twice.

File: spec/renderer/renderer.spec.js [hunk 3]
```diff
@@ -96,5 +114,58 @@ describe('The Renderer Class', function() {
             return d.letters;
         }
     });
+    
+    it('should be able to nest conditionals and loops', function() {
```
Comment (sdtsui) [inline spec/renderer/renderer.spec.js]: like above, would be clearer if conditionals were tested separate from loops. not clear on how loops (each/xeach) work right now. will continue reviewing.

File: spec/renderer/renderer.spec.js (hunk 1 above)
Comment (chikeichan) [inline spec/renderer/renderer.spec.js]: What do you think about deprecating removeClass at all? It's kinda pointless having a removeClass method in a templating API. Why would people want to remove class from a template? They can just not add the class in the first place, right?

I will add spec for first edge case. I know i have code that will prevent the same class from being added twice. Good idea to cover it with a spec.

File: spec/renderer/renderer.spec.js (hunk 2 above)
Comment (chikeichan) [inline spec/renderer/renderer.spec.js]: good idea. Will add

File: spec/renderer/renderer.spec.js (hunk 3 above)
Comment (chikeichan) [inline spec/renderer/renderer.spec.js]: will add separate spec for conditionals and loops, and still keeping these two.

**Pull request closed without merging**